app/src/main/kotlin/com/github/gotify/service/WebSocketService.kt

```diff
@@ -312,13 +313,26 @@ internal class WebSocketService : Service() {
             PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
         )
 
-        val b = NotificationCompat.Builder(
-            this,
-            NotificationSupport.convertPriorityToChannel(priority)
-        )
+        val channelId: String
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O &&
+            NotificationSupport.areAppChannelsRequested(this)
+        ) {
+            channelId = NotificationSupport.getChannelID(priority, appId.toString())
+            NotificationSupport.createChannelIfNonexistent(
+                this,
+                (this.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager),
+                appId.toString(),
+                appId.toString(),
+                channelId
+            )
```

Comment:
I think the method signature of the createChannelIfNonexistent could be improved, only pass this, appId & channelId.

Reply:
Maybe custom changes from a user will be discarded if the channel will be added if it already exist. But not sure.

Reply:
I suppose that we can save time and battery if we don't always try to create channels that already exist.
I would keep that

Reply:
So you would update the channels everytime a notification is received to keep the name and the importance up to date?  
I see your point.

I mean as a power user I sometimes receive 20 - 30 messages a day.  
Do we want to always update the channels? Honestly I don't know.

@jmattheis What do you think? You decide

Reply:
I mean, the apps won't be refreshed when receiving a new message. The notification channels should only be updated when the apps are refetched in the WebSocketService.

Reply:
>  Didn't look through the code in detail but won't there be just 1 channel per app or will there be multiple channels per app with different importances?

There will be four channels per gotify application.

Reply:
There are multiple channels per app to keep the given importance functional. Otherwise it would be ignored if there was only one channel.

I'm confused. In the beginning you said you would drop the check and always update the notification channels, now you are suggesting the opposite?

Reply:
> I mean, the apps won't be refreshed when receiving a new message. The notification channels should only be updated when the apps are refetched in the WebSocketService.

Right, so forcing an update on every received message would not be necessary, right?

Reply:
Yes

Reply:
> Isn't that a bit too much maybe. If one wanted to differentiate by priority, he could rather stick to the current implementation that will still be available instead of using the app channels implementation?

Mhmm, not sure. I'm leaning towards your suggestion, but I dislike that the priority would be ignored.

Reply:
True, oke, let's only do one channel per application. We probably need some kind of explanation of what this setting does, so that a user understands that priorities won't work when this setting is enabled.

Reply:
I like the approach of separate channels per app.

For example: I would use the separate channels per app to be able to mute single apps OR to mute only specific channels of a single app. That would not be possible anymore if you merge the channels into one.

Those, who don't care about the importance levels, can stil use this feature and ignore the separate channels. Everything will work the same way and you can still mute all channels of an app at once.

Reply:
Okay, then let's leave it as is with four channels per application.

Reply:
Coming back to the reason this review was opened, the method signature,  
Resolved in 5399d003e40fc1068e5f69814bbf2b120cc44096